types/v56/nsxt_types.go

```diff
@@ -42,6 +58,67 @@ type OpenAPIEdgeGateway struct {
 	// supported for VMC. If nothing is set, the default is 192.168.255.225/27. The DHCP listener IP network is on
 	// 192.168.255.225/30. The DNS listener IP network is on 192.168.255.228/32. This field cannot be updated.
 	ServiceNetworkDefinition string `json:"serviceNetworkDefinition,omitempty"`
+
+	// UsingIpSpace is a boolean flag to indicate whether the edge gateway is using IP space or not.
+	UsingIpSpace *bool `json:"usingIpSpace,omitempty"`
+}
+
+// DeallocateIpCount modifies the structure to deallocate IP addresses from the Edge Gateway
+// uplinks.
+//
+// Note. Use QuickAddAllocatedIPCount field in the uplink structure to leverage VCD API directly
+// for allocating IP addresses.
+func (egw *OpenAPIEdgeGateway) DeallocateIpCount(deallocateIpCount int) error {
```

Comment:
Why is this method not in the govcd package?
I am concerned about the dependency from package `util`, which has never been used in `types` before

Reply:
I initially put this here because it does not perform any API operations (only modifies structure) however, I didn't like that myself too much. Moved it


Now we have two `*NsxtEdgeGateway` pointer receiver functions:
* DeallocateIpCount           (does only mutation of structure)
* QuickDeallocateIpCount (uses the above function and performs update API operation)

Both of them have comments to stress the difference between them